dispatch/dispatch.go

```diff
@@ -175,28 +177,30 @@ func (d *Dispatcher) run(it provider.AlertIterator) {
 			}
 			d.metrics.processingDuration.Observe(time.Since(now).Seconds())
 
-		case <-cleanup.C:
-			d.mtx.Lock()
-
-			for _, groups := range d.aggrGroupsPerRoute {
-				for _, ag := range groups {
-					if ag.empty() {
-						ag.stop()
-						delete(groups, ag.fingerprint())
-						d.aggrGroupsNum--
-						d.metrics.aggrGroups.Dec()
-					}
-				}
-			}
-
-			d.mtx.Unlock()
-
+		case <-maintenance.C:
+			d.doMaintenance()
 		case <-d.ctx.Done():
 			return
 		}
 	}
 }
 
+func (d *Dispatcher) doMaintenance() {
+	d.mtx.Lock()
```

Comment:
nit: we tend to have this prefernece in the AM - it made sense before to have them split but not now.

```suggestion
	d.mtx.Lock()
	defer d.mtx.Unlock()
```

Reply:
Will fix! 👍